kyaml/yaml/merge2/merge2.go

```diff
@@ -45,9 +45,9 @@ func MergeStrings(srcStr, destStr string, infer bool, mergeOptions yaml.MergeOpt
 }
 
 type Merger struct {
-	// for forwards compatibility when new functions are added to the interface
```

Comment:
I feel this comment means that when we need to add some field in this struct.
So, I think this comment is still in the right place.

Reply:
Hi @koba1t , and thanks for your review! I think I disagree with you on your point:
As far as I understand, the comment is applying to the pattern that is used to ensure that if functions are added to the `walk.Visitor` interface, the struct will still implement all of them (if not, the build will failed because of the line `var _ walk.Visitor = Merger{}`).
And also the comment refers to the *interface*, which `Merger` is not (it is a struct), so again, the comment should apply to the line mentioned above.
So if a new function is added on the `walk.Visitor` interface, the line `var _ walk.Visitor = Merger{}` will ensure that the `Merger` struct is still (forward-)compatible with the interface, which is basically what the comment is saying.
WDYT?

Reply:
Oh, Thanks for describing this code. I misunderstood `walk.Visitor` as not an interface.
I agree with what you are doing now.